### ISSUE TITLE
Span across multiple displays

### ISSUE THREAD

**Issue by ecocode** [danyspin97/wpaperd#2]
Hi,

This is nice.  Would you consider adding feature to span a wallpaper across multiple outputs ?

**Comment by danyspin97** [comment]
Hi!

This is a good idea. However, it seems to me that it isn't easy to implement. I have to think a bit on how to get the outputs and their position.

I am also thinkg about the assoctiated configuration in `output.conf`. Maybe adding a `group` key for each output seems a good way to go. What do you think?

**Comment by progandy** [comment]
> This is a good idea. However, it seems to me that it isn't easy to implement. I have to think a bit on how to get the outputs and their position.

You'd have to use the privileged ```wlr_output_management_unstable_v1``` protocol for a generic wlroots implementation (e.g. [wlr-randr](https://sr.ht/~emersion/wlr-randr/) relies on it) or swaymsg for sway only. 
```wl_output::geometry``` [does not work in wlroots](https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/1610)

**Comment by danyspin97** [comment]
That would be great, thank you! Let me know if you need any help or review :)

**Comment by Shinyzenith** [comment]
So a small update on my message that I posted above. I needed a similar algorithm of figuring out display bounds on wlroots which I haven't figured out yet (after pouring approximately 9 hours into wrong implementations).. this is taking wayyy longer than I expected and I'm in touch with a smithay dev regarding it. Hoping to find a solution soon.

Cheers.

**Comment by danyspin97** [comment]
Thank you for the update and for working on this 😄 
I think having a look at how [wdisplays](https://github.com/artizirk/wdisplays) get this info would be a good start. Also consider that smithay-client-toolkit is currently being refactored and reworked for the next update, so wpaperd would also require to be refactored and updated as well.

**Comment by Shinyzenith** [comment]
> Thank you for the update and for working on this smile I think having a look at how [wdisplays](https://github.com/artizirk/wdisplays) get this info would be a good start. Also consider that smithay-client-toolkit is currently being refactored and reworked for the next update, so wpaperd would also require to be refactored and updated as well.

absolutely, wayland-rs 0.3 will be a huge breaking change... I will try and help out with the refactor.
As for the display bounds I tried using smithay_client_toolkit::output::OutputInfo offset field.

**Comment by CardboardTurkey** [comment]
Any progress on this?

**Comment by danyspin97** [comment]
No progress yet unfortenately. wpaperd has been refactored, but before implementing this feature I'd like to create a `wpaper` command line program to get the info about the current system and have a `next` command.
